### ISSUE TITLE
Keychain exception on iOS after upgrading to 4.3.0

### ISSUE THREAD

**Issue by ddomengeaux** [realm/realm-dotnet#1956]
## Goals
 Update nuget to 4.3.0

## Expected Results

Not to crash

## Actual Results

RealmException: Keychain returned unexpected status code: -34018 is thrown

```
Realms.Exceptions.RealmException: Keychain returned unexpected status code: -34018
  at Realms.NativeException.ThrowIfNecessary (System.Func`2[T,TResult] overrider) [0x0000a] in C:\jenkins\workspace\realm_realm\Realm\Realm\Native\NativeException.cs:57
  at Realms.Sync.SharedRealmHandleExtensions.Configure (System.Nullable`1[T] userPersistenceMode, System.Byte[] encryptionKey, System.Boolean resetMetadataOnError, System.String basePath) [0x0007c] in C:\jenkins\workspace\realm_realm\Realm\Realm\Handles\SharedRealmHandleExtensions.cs:234
  at Realms.Sync.SharedRealmHandleExtensions.DoInitialFileSystemConfiguration () [0x0000d] in C:\jenkins\workspace\realm_realm\Realm\Realm\Handles\SharedRealmHandleExtensions.cs:198
  at Realms.Sync.User.get_Current () [0x00000] in C:\jenkins\workspace\realm_realm\Realm\Realm\Sync\User.cs:49
  at App1.MainPage.OnAppearing () [0x00008] in /Users/danield/Projects/realmBug/App1/App1/App1/MainPage.xaml.cs:22
  at Xamarin.Forms.Page.SendAppearing () [0x00045] in D:\a\1\s\Xamarin.Forms.Core\Page.cs:406
  at Xamarin.Forms.Platform.iOS.PageRenderer.ViewDidAppear (System.Boolean animated) [0x00045] in D:\a\1\s\Xamarin.Forms.Platform.iOS\Renderers\PageRenderer.cs:202
  at at (wrapper managed-to-native) UIKit.UIApplication.UIApplicationMain(int,string[],intptr,intptr)
  at UIKit.UIApplication.Main (System.String[] args, System.IntPtr principal, System.IntPtr delegate) [0x00005] in /Library/Frameworks/Xamarin.iOS.framework/Versions/13.10.0.17/src/Xamarin.iOS/UIKit/UIApplication.cs:86
  at UIKit.UIApplication.Main (System.String[] args, System.String principalClassName, System.String delegateClassName) [0x0000e] in /Library/Frameworks/Xamarin.iOS.framework/Versions/13.10.0.17/src/Xamarin.iOS/UIKit/UIApplication.cs:65
  at App1.iOS.Application.Main (System.String[] args) [0x00001] in /Users/danield/Projects/realmBug/App1/App1/App1.iOS/Main.cs:16
```

## Steps to Reproduce

Created new xamarin forms application and added realm nuget 4.3.0 and simply calling User.Current is enough to crash.

## Code Sample

https://github.com/ddomengeaux/realm-keychain-bug.git

## Version of Realm and Tooling

- Realm Object Server Version: 3.28.2
- Flavor: Realm Cloud
- Client SDK Version: 4.3.0
- Client OS & Version: iOS 13.3

**Comment by fealebenpae** [comment]
I'll try to reproduce and fix this ASAP.

**Comment by RZiegler979** [comment]
I have the same problem on a windows service I am trying to run and a windows forms app. I can provide code examples if they would help.

**Comment by fealebenpae** [comment]
@RZiegler979 the Keychain error is exclusive to iOS - perhaps what you're seeing is something else? I suggest opening a new issue with a stack trace and other exception information.

**Comment by RZiegler979** [comment]
My mistake. Same error message, happened around the upgrade to 4.3.0, will create new issue.

**Comment by VagueGit** [comment]
@RZiegler979 is it the same as [Upgrade from 4.2.0 to 4.3.0 throws Metadata Realm encryption was specified, but no encryption key was provided #1960](https://github.com/realm/realm-dotnet/issues/1960)

**Comment by RZiegler979** [comment]
I thought so, but @fealebenpae stated that linked problem was the same as this one, and this one isn't the same as mine, because it's from IOS, so I created a new bug as he advised. https://github.com/realm/realm-dotnet/issues/1967

**Comment by fealebenpae** [comment]
@VagueGit can you please try and see if the `Realms.Sync.SyncConfigurationBase.Configure()` workaround I posted in https://github.com/realm/realm-dotnet/issues/1967#issuecomment-595753987 helps? If it does, I think you might be getting the same issue as @RZiegler979.

**Comment by VagueGit** [comment]
@fealebenpae as encrypted realms are now the default, perhaps Realm docs should include guidance on migrating unencrypted realms to encrypted.

Also there was no mention of this as a breaking change in the release notes.

We released our app in preview mode when Realm Cloud was first released. We used unencrypted Realms for simplicity. However we now have many users who have migrated their businesses from the release version of our app (not Realm) to the preview version on Realm Cloud, so we need guidance on how to migrate these realms from unencrypted to encrypted.

If necessary we can wait until we migrate to Atlas if that simplifies things.

_Originally posted by @VagueGit in https://github.com/realm/realm-dotnet/issues/1967#issuecomment-596793351_
